src/azul/openapi/schema.py

```diff
@@ -117,6 +122,90 @@ def array(item: TYPE, *items: TYPE, **kwargs):
     return array_type(make_type(item), *map(make_type, items), **kwargs)
 
 
+def enum_type(items: Iterable) -> JSON:
+    """
+    Specify allowed values for a parameter schema.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(enum_type({1, 2}))
+    {
+        "type": "integer",
+        "format": "int64",
+        "enum": [
+            1,
+            2
+        ]
+    }
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(enum_type({'foo', 'bar'}))
+    {
+        "type": "string",
+        "enum": [
+            "bar",
+            "foo"
+        ]
+    }
+    """
+    items = sorted(items)
+    type_ = one(set(map(type, items)))
+    return dict(make_type(type_), enum=items)
+
+
+def path_param(name: str, type_: TYPE, description: Optional[str] = None) -> JSON:
+    """
+    Returns a JSON `parameters` entry for an endpoint parameter specified via
+    the URL path.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(path_param('foo', int))
+    {
+        "name": "foo",
+        "in": "path",
+        "required": true,
+        "schema": {
+            "type": "integer",
+            "format": "int64"
+        }
+    }
+    """
+    # Path params must have required=True
+    return make_param(name, 'path', type_, required=True, description=description)
+
+
+def query_param(name: str, type_: Union[TYPE, optional], description: Optional[str] = None) -> JSON:
+    """
+    Returns a JSON `parameters` entry for an endpoint parameter specified via
+    the URL query parameters.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(query_param('foo', optional(int)))
+    {
+        "name": "foo",
+        "in": "query",
+        "required": false,
+        "schema": {
+            "type": "integer",
+            "format": "int64"
+        }
+    }
+    """
+    is_optional = isinstance(type_, optional)
+    if is_optional:
+        type_ = type_.property
+    return make_param(name, 'query', type_, required=not is_optional, description=description)
+
+
+def header_response(type_: TYPE, description: Optional[str] = None) -> JSON:
+    """
+    Returns the schema and description for a response header.
+    """
+    return {
+        'schema': make_type(type_),
+        **({} if description is None else {'description': description})
```

Comment:
This conditional can be eliminated simply by allowing and interpolating `**kwargs` like some other helpers do. 

This is not supposed to be an abstraction over OpenAPI spec, but merely a way to reduce its verbosity when specified as a JSON literal.